src/System.IO.Abstractions.TestingHelpers/MockFile.cs

```diff
@@ -539,6 +541,12 @@ public override Stream Open(string path, FileMode mode, FileAccess access)
 
             var mockFileData = mockFileDataAccessor.GetFile(path);
             mockFileData.CheckFileAccess(path, access);
+            var timeAdjustments = TimeAdjustments.LastAccessTime;
+            if (access != FileAccess.Read)
```

Comment:
Perhaps use the explicit check if we are dealing with any write, e.g. `ReadWrite` or `Write`

```suggestion
            if (access.HasFlag(FileAcess.Write))
```

Reply:
Implemented as suggested